### ISSUE TITLE
Add support for Microsoft Speech API version 5 (SAPI5) 64-bit voices

### ISSUE THREAD

**Issue by tmthywynn8** [nvaccess/nvda#5905]
Currently, when running under 64-bit operating systems, NVDA only supports voices under the HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\SPEECH\Voices\Tokens registry key. In other words, the voice list for NVDA is only populated from the subkeys under that node.

Request: NVDA should also be able to populate voices under the HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\SPEECH\Voices\Tokens. The problems with this request are:
  1. there may be duplicate entries for a particular voice if said voice exists under both the 32 and 64-bit nodes. Solutions: Only display either the 32 or 64-bit voice, or create a separate synthesizer for 64-bit voices (possibly named "Microsoft Speech API version 5 (64-bit)". The second solution is what I believe the Freedom Scientific's JAWS screen reader does.
  2. As I think NVDA is a 32-bit application, it may not even e able to view entries outside the Wow6432Node in the registry. In that case, my request is not feasible and should be tagged as "cantfix" or what have you.

**Comment by jcsteh** [comment]
From what I've seen, 64 bit voice packages usually include a 32 bit 
engine as well. What voices have you seen where this isn't the case?

A 32 bit app can read the 64 bit registry. The bigger problem is that a 
32 bit app can't load a 64 bit dll. I once tried getting COM to load 64 
bit voices out-of-process, but this didn't seem to work for some reason, 
perhaps because SAPI5 is declared as in-process; I'm not sure.

**Comment by tmthywynn8** [comment]
I actually am not sure. I thought I recall an older Vietnamese voice doing this, but I don't think it even had 64-bit-compiled executables or DLLs. The installer could have imported a .reg file to manually create the voice (incorrectly at that), but again, not sure. Will download and test in a few days to make sure I'm not imagining things.

JAWS does have a 64-bit option when choosing SAPI5, so I suppose this must've been done for at least one TTS vender. When seeing the 64-bit TTS choice yesterday, I created the ticket just in case this problem ever arises for someone else. I could, of course, fake the problem by using the registry keys for the Speech OneCore voices, but that certainly isn't a real-world situation.

Let me test the older Vietnamese voice and get back to you. If I misremembered things, we could just close the ticket as there are no currently known cases of a voice only for 64-bit programs.

**Comment by tmthywynn8** [comment]
I did test the voice I mentioned, and it does properly integrate said voice in the proper 32-bit node in the registry. It was a definite case of poor recall. Though 1) JAWS has support for 64-bit SAPI5 voices, and 2) the Vietnamese voices does have issues on modern operating systems, the problems had nothing to do with 64-bit SAPI5.

This ticket will now be closed, unless a use case ever arises wherein a SAPI5 voice is 64-bit only.
